packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -1275,6 +1279,23 @@ function prepareFreshStack(root, expirationTime) {
   }
 }
 
+export function unstable_avoidThisRender(value: Thenable) {
+  invariant(
+    workInProgressRoot !== null &&
+      workInProgress !== null &&
```

Comment:
Pass `workInProgress` as an argument from the dispatcher method, where it's guaranteed to be defined, since the method is only accessible in the render phase.